packages/story-editor/src/app/story/effects/useHistoryEntry.ts

```diff
@@ -51,30 +53,38 @@ const ELEMENT_PROPS_TO_IGNORE = [
 ];
 
 // Record any change to core variables in history (history will know if it's a replay)
-function useHistoryEntry({ story, current, pages, selection, capabilities }) {
+function useHistoryEntry({
+  story,
+  current,
+  pages,
+  selection,
+  capabilities,
+}: HistoryEntry) {
   const {
     state: { currentEntry },
     actions: { stateToHistory },
   } = useHistory();
 
-  const currentHistoryEntryRef = useRef();
+  const currentHistoryEntryRef = useRef<HistoryEntry | null>(null);
   useEffect(() => {
     if (currentEntry) {
       currentHistoryEntryRef.current = structuredClone(currentEntry);
     }
   }, [currentEntry]);
 
-  const currentPageIndexRef = useRef();
-  const selectedElementIdsRef = useRef();
+  const currentPageIdRef = useRef<string | null>(null);
+  const selectedElementIdsRef = useRef<string[]>([]);
   useEffect(() => {
-    currentPageIndexRef.current = current;
+    currentPageIdRef.current = current;
     selectedElementIdsRef.current = selection;
   }, [current, selection]);
 
-  const deleteKeysFromPages = (list) => {
+  const deleteKeysFromPages = (list: Page[]) => {
     // Create a copy of the list not to influence the original.
     return structuredClone(list).map((page) => {
-      page.elements.forEach(deleteNestedKeys(ELEMENT_PROPS_TO_IGNORE));
+      page.elements.forEach((element) =>
+        deleteNestedKeys(ELEMENT_PROPS_TO_IGNORE)({ ...element })
```

Comment:
🤨

Reply:
That's because otherwise it's not clear that `Element` indexes are `string`s 🤷 
Not the nicest solution.